generator/sidekick/config.go

```diff
@@ -22,9 +29,77 @@ type Config struct {
 }
 
 type GeneralConfig struct {
+	Language            string `toml:"language"`
+	TemplateDir         string `toml:"template-dir"`
 	SpecificationFormat string `toml:"specification-format"`
 	SpecificationSource string `toml:"specification-source"`
 	ServiceConfig       string `toml:"service-config"`
-	Language            string `toml:"language"`
-	TemplateDir         string `toml:"template-dir"`
+}
+
+func LoadRootConfig(filename string) (*Config, error) {
+	var config Config
+	if contents, err := os.ReadFile(filename); err == nil {
+		err = toml.Unmarshal(contents, &config)
+		if err != nil {
+			return nil, fmt.Errorf("error reading top-level configuration: %w", err)
+		}
+	}
+	if config.Codec == nil {
```

Comment:
nit: I'm not sure why these need to be initialized here as opposed to just leaving them nil, but if you need to I would do it on line 40

```
config := &Config{
   Codec: map[string]string{},
   Source: map[string]string{}
}
```

Reply:
Moved the initialization. Also took the opportunity to add some tests.

Is it idiomatic to leave maps `nil`? And then one checks them before writing to them?